src/lsm/lsm_merge.c

```diff
@@ -255,21 +282,10 @@ __lsm_merge_span(WT_SESSION_IMPL *session, WT_LSM_TREE *lsm_tree,
 		if (oldest_gen - youngest_gen > max_gap)
 			break;
 
-		/*
-		 * If we have enough chunks for a merge and the next chunk is
-		 * in too high a generation, stop.
-		 */
-		if (nchunks >= merge_min) {
-			previous = lsm_tree->chunk[start_chunk];
-			max_gen = youngest->generation + max_gap;
-			if (previous->generation <= max_gen &&
-			    chunk->generation > max_gen)
-				break;
-		}
-
 		F_SET(chunk, WT_LSM_CHUNK_MERGING);
 		record_count += chunk->count;
 		--start_chunk;
+		++nchunks;
```

Comment:
Initially, ```nchunks = (end_chunk + 1) - start_chunk```. Here start_chunk is updated, hence nchunks should be updated as well. Later nchunks will be used, it should reflect the most recent data.